model/DeliveryFactory.php

```diff
@@ -230,4 +229,69 @@ protected function createDeliveryResource(
 
         return $compilationInstance;
     }
+
+    private function createNamespacedDeliveryResource(
+        KernelClass $deliveryClass,
+        array $additionalParameters
+    ): KernelResource {
+        $deliveryId = trim($additionalParameters[DeliveryAssemblyService::PROPERTY_ASSESSMENT_PROJECT_ID] ?? '');
+
+        if (!$deliveryId) {
+            throw new RuntimeException(
+                sprintf('%s must not be empty.', DeliveryAssemblyService::PROPERTY_ASSESSMENT_PROJECT_ID)
+            );
+        }
+
+        $this->getDeliveryDeleteService()
+            ->execute(
+                $this->createDeliverDeleteRequest($deliveryId)
+            );
+
+        $delivery = $deliveryClass->getResource($this->createNamespacedDeliveryId($deliveryId));
+
```

Comment:
the code looks good but i would remove the extra \n

Reply:
I would rather keep it to separate the 4 blocks from each other:
1. Resource retrieval
2. Resource's properties modification
3. Event triggering
4. return value